core/client/src/main/java/alluxio/client/file/FileSystemUtils.java

```diff
@@ -145,59 +136,29 @@ public static boolean waitCompleted(final FileSystem fs, final AlluxioURI uri,
    *
    * @param fs {@link FileSystem} to carry out Alluxio operations
    * @param uri the uri of the file to persist
-   * @param status the status info of the file
-   * @return the size of the file persisted
    * @throws IOException if an I/O error occurs
    * @throws FileDoesNotExistException if the given file does not exist
    * @throws AlluxioException if an unexpected Alluxio error occurs
    */
-  public static long persistFile(FileSystem fs, AlluxioURI uri, URIStatus status)
-      throws IOException, FileDoesNotExistException, AlluxioException {
-    // TODO(manugoyal) move this logic to the worker, as it deals with the under file system
-    Closer closer = Closer.create();
-    long ret;
+  public static void persistFile(final FileSystem fs, final AlluxioURI uri)
+      throws AlluxioException, IOException {
+    FileSystemContext context = FileSystemContext.INSTANCE;
+    FileSystemMasterClient client = context.acquireMasterClient();
     try {
-      OpenFileOptions options = OpenFileOptions.defaults().setReadType(ReadType.NO_CACHE);
-      FileInStream in = closer.register(fs.openFile(uri, options));
-      AlluxioURI dstPath = new AlluxioURI(status.getUfsPath());
-      UnderFileSystem ufs = UnderFileSystem.Factory.get(dstPath.toString());
-      // Create ancestor directories from top to the bottom. We cannot use recursive create parents
-      // here because the permission for the ancestors can be different.
-      Stack<Pair<String, MkdirsOptions>> ufsDirsToMakeWithOptions = new Stack<>();
-      AlluxioURI curAlluxioPath = uri.getParent();
-      AlluxioURI curUfsPath = dstPath.getParent();
-      // Stop at the Alluxio root because the mapped directory of Alluxio root in UFS may not exist.
-      while (!ufs.isDirectory(curUfsPath.toString()) && curAlluxioPath != null) {
-        URIStatus curDirStatus = fs.getStatus(curAlluxioPath);
-        Permission perm = new Permission(curDirStatus.getOwner(), curDirStatus.getGroup(),
-            (short) curDirStatus.getMode());
-        ufsDirsToMakeWithOptions.push(new Pair<>(curUfsPath.toString(),
-            MkdirsOptions.defaults().setCreateParent(false).setPermission(perm)));
-
-        curAlluxioPath = curAlluxioPath.getParent();
-        curUfsPath = curUfsPath.getParent();
-      }
-      while (!ufsDirsToMakeWithOptions.empty()) {
-        Pair<String, MkdirsOptions> ufsDirAndPerm = ufsDirsToMakeWithOptions.pop();
-        if (!ufs.mkdirs(ufsDirAndPerm.getFirst(), ufsDirAndPerm.getSecond())) {
-          throw new IOException("Failed to create " + ufsDirAndPerm.getFirst() + " with permission "
-              + ufsDirAndPerm.getSecond().toString());
+      client.scheduleAsyncPersist(uri);
+      CommonUtils.waitFor("Wait for the file to be persisted", new Function<Void, Boolean>() {
+        @Override
+        public Boolean apply(Void input) {
+          try {
+            return fs.getStatus(uri).isPersisted();
```

Comment:
This is going to send an RPC to the master every 20ms, do you mind toning it down?